File: fix_all.js
const fs = require('fs');

let content = fs.readFileSync('/tmp/perfect_original_CampPointsTracker.jsx', 'utf8');

// 1. Update CONTINENT_ORDER
content = content.replace(
    "const CONTINENT_ORDER = ['North America', 'South America', 'Europe', 'Africa', 'Asia & Oceania'];",
    "const CONTINENT_ORDER = ['America', 'Europe', 'Africa', 'Asia & Oceania'];"
);

// 2. Fix colors and continents in TEAM_META
const updates = {
    'USA': { colorOverride: '#B31942', continent: 'America' }, // B31942 is close enough to USAColors, wait, user said USAColors. The original was fine for USA. Let's just set continent: 'America'.
    'Mexico': { colorOverride: '#8FBC8F', continent: 'America' }, // Sage Green
    'Argentina': { colorOverride: '#ADD8E6', continent: 'America' }, // Light Blue
    'Brazil': { colorOverride: '#008000', continent: 'America' }, // Green
    'Colombia': { colorOverride: '#FFFF00', continent: 'America' }, // Yellow
    'England': { colorOverride: '#FFFFFF', continent: 'Europe' }, // White
    'France': { colorOverride: '#00008B', continent: 'Europe' }, // Dark Blue
    'Spain': { colorOverride: '#FF0000', continent: 'Europe' }, // Red
    'Germany': { colorOverride: '#A52A2A', continent: 'Europe' }, // Brown
    'Egypt': { colorOverride: '#FFD700', continent: 'Africa' }, // Gold
    'South Africa': { colorOverride: '#000000', continent: 'Africa' }, // Black
    'Ghana': { colorOverride: '#FFA500', continent: 'Africa' }, // Orange
    'Morocco': { colorOverride: '#006400', continent: 'Africa' }, // Dark kidi Green
    'Japan': { colorOverride: '#8B008B', continent: 'Asia & Oceania' }, // Dark Pink
    'South Korea': { colorOverride: '#FFB6C1', continent: 'Asia & Oceania' }, // Light Pink
    'New Zealand': { colorOverride: '#E0B0FF', continent: 'Asia & Oceania' }, // Light Purple
    'Australia': { colorOverride: '#800080', continent: 'Asia & Oceania' } // Purple
};

// Replace 'North America' and 'South America' with 'America'
content = content.replace(/continent: 'North America'/g, "continent: 'America'");
content = content.replace(/continent: 'South America'/g, "continent: 'America'");

// Apply color overrides
for (const [team, meta] of Object.entries(updates)) {
    // Regex to match the team line
    const regex = new RegExp(`('${team}':\\s*\\{.*?)(color_hex:\\s*'[^']+'|colorOverride:\\s*'[^']+')?(,\\s*dividerClass.*?)?(\\},?)$`, 'gm');
    // Wait, regex with .*? can still fail on SVG if not careful.
    // Let's just do simple string replacements for colors!
}

fs.writeFileSync('/tmp/fix_all.js_out', content);

File: fix_all.py
import re

with open('/tmp/perfect_original_CampPointsTracker.jsx', 'r') as f:
    content = f.read()

# 1. Update CONTINENT_ORDER
content = content.replace(
    "const CONTINENT_ORDER = ['North America', 'South America', 'Europe', 'Africa', 'Asia & Oceania'];",
    "const CONTINENT_ORDER = ['America', 'Europe', 'Africa', 'Asia & Oceania'];"
)

# 2. Continents: 'North America' and 'South America' -> 'America'
content = content.replace("continent: 'North America'", "continent: 'America'")
content = content.replace("continent: 'South America'", "continent: 'America'")

# 3. Colors
colors = {
    'Mexico': '#8FBC8F',
    'Argentina': '#ADD8E6',
    'Brazil': '#008000',
    'Colombia': '#FFFF00',
    'England': '#FFFFFF',
    'France': '#00008B',
    'Spain': '#FF0000',
    'Germany': '#A52A2A',
    'Egypt': '#FFD700',
    'South Africa': '#000000',
    'Ghana': '#FFA500',
    'Morocco': '#006400',
    'Japan': '#8B008B',
    'South Korea': '#FFB6C1',
    'New Zealand': '#E0B0FF',
    'Australia': '#800080'
}

for team, color in colors.items():
    # If the team has a colorOverride, replace it
    if f"'{team}':" in content:
        # Find the line that starts with 'team':
        # Since some teams have multiline strings, we can't just replace line by line easily without parsing.
        # But we can look for `'Team': {` and inject colorOverride right after `{ `
        content = re.sub(
            f"('{team}': \\{{)", 
            f"\\1 colorOverride: '{color}',", 
            content
        )

# Remove color_hex, dividerClass, topBlockClass if they conflict
# Actually we can just leave them if we want, colorOverride takes precedence in our updated enrichedTeams.

# 4. Remove Canada, Panama, Uruguay
# We can do this by regexing the exact lines because they are single lines!
content = re.sub(r"^\s*'Canada': \{.*?\},?\n?", "", content, flags=re.MULTILINE)
content = re.sub(r"^\s*'Panama': \{.*?\},?\n?", "", content, flags=re.MULTILINE)
content = re.sub(r"^\s*'Uruguay': \{.*?\},?\n?", "", content, flags=re.MULTILINE)

# 5. Append missing teams
# We will append them right before the closing brace of TEAM_META.
# Look for "};" right after "Asia & Oceania" block.
missing_teams = """
    'Portugal': { continent: 'Europe', flag: '🇵🇹', rank: 4, code: 'POR', colorOverride: '#800000' },
    'Senegal': { continent: 'Africa', flag: '🇸🇳', rank: 5, code: 'SEN', colorOverride: '#C0C0C0' },
    'Uzbekistan': { continent: 'Asia & Oceania', flag: '🇺🇿', rank: 5, code: 'UZB', colorOverride: '#008080' }
"""
content = content.replace("};\n\nconst CONTINENT_ORDER", missing_teams + "};\n\nconst CONTINENT_ORDER")

# 6. Inject the sync logic and update enrichedTeams
sync_code = """
    // DB Sync
    useEffect(() => {
        const syncDB = async () => {
            const { data: existing, error: fetchErr } = await supabase.from('camp_teams').select('*');
            if (fetchErr) return;
            const newNames = Object.keys(TEAM_META);
            const existingNames = existing.map(t => t.name);
            
            // Delete old
            for (let t of existing) {
                if (!newNames.includes(t.name)) {
                    await supabase.from('camp_teams').delete().eq('id', t.id);
                }
            }
            // Insert new
            for (let name of newNames) {
                if (!existingNames.includes(name)) {
                    await supabase.from('camp_teams').insert({ name, points: 0, color_hex: TEAM_META[name]?.colorOverride || '#000000' });
                }
            }
        };
        syncDB();
    }, []);
"""
content = content.replace(
    "const [activeView, setActiveView] = useState('main'); // 'main', 'podium', 'continents'",
    "const [activeView, setActiveView] = useState('main'); // 'main', 'podium', 'continents'\n" + sync_code
)

enriched_teams_code = """
    const enrichedTeams = teams.filter(team => TEAM_META[team.name]).map(team => ({
        ...team,
        color_hex: TEAM_META[team.name]?.colorOverride || team.color_hex,
        continent: TEAM_META[team.name]?.continent || 'Other',
        flag: TEAM_META[team.name]?.flag || '🏳️',
        rank: TEAM_META[team.name]?.rank || 99,
        bgStyle: TEAM_META[team.name]?.bgStyle || null,
        bgPosition: TEAM_META[team.name]?.bgPosition || '0px 0px',
        code: TEAM_META[team.name]?.code || team.name.substring(0, 3).toUpperCase(),
        cups: TEAM_META[team.name]?.cups || 0,
        topBlockStyle: TEAM_META[team.name]?.topBlockStyle || null,
        dividerClass: TEAM_META[team.name]?.dividerClass || '',
        topBlockClass: TEAM_META[team.name]?.topBlockClass || ''
    }));
"""
content = re.sub(
    r"    const enrichedTeams = teams\.map.*?\}\)\);",
    enriched_teams_code.strip(),
    content,
    flags=re.DOTALL
)

with open('src/pages/CampPointsTracker.jsx', 'w') as f:
    f.write(content)
